ENH: Add precomputed_slices.

ssanderson opened the pull request.

Commit:
ENH: Add precomputed_slices.

llllllllll commented on codetransformer/transformers/precomputed_slices.py:
```diff
@@ -0,0 +1,20 @@
+"""
```
maybe put this in constants?

ssanderson replied:
I think this is different enough from constants to have its own file.

llllllllll replied:
idk, it is about changing the constants of a function. I don't have a strong preference but I think they are related.

ssanderson replied:
I think of `asconstants` as being about injecting your own custom constants.

I'd be more inclined to put this in something like a top-level `optimizers` package that has transformers that do semantics-preserving optimizaitons.

llllllllll replied:
Okay, I think that we should do something like `optimizations` when we add more.

Commit:
ENH: Add Code.from_pyfunc.

Commit:
BUG: Fix duplicated varnames.

Arguments with STORE_FASTS to their names were being duplicated by
Code.varnames.

Commit:
TEST: Add another test for precomputed_slices.
